app.js

```diff
@@ -29,33 +30,59 @@ const sessionParms = {
 };
 
 if (app.get("env") === "production") {
-  app.set("trust proxy", 1); // trust first proxy
-  sessionParms.cookie.secure = true; // serve secure cookies
+  app.set("trust proxy", 1);
+  sessionParms.cookie.secure = true;
 }
 
+app.use(cookieParser()); // Must be before CSRF middleware
 app.use(session(sessionParms));
 
 const passport = require("passport");
 const passportInit = require("./passport/passportInit");
-
 passportInit();
 app.use(passport.initialize());
 app.use(passport.session());
 
 app.use(require("connect-flash")());
-
 app.use(require("./middleware/storeLocals"));
+
+//  CSRF Protection Middleware (Must be after session and body-parser)
+app.use(csurf());
+
+//  Middleware to pass CSRF token to views
+app.use((req, res, next) => {
+  res.locals._csrf = req.csrfToken();  // Make sure CSRF token is available in views
```

Comment:
With this being set in your middleware, you shouldn't need to explicitly pass your `_csrf` separately as you do below.